int32 key_id
int32 from_vertex
int32 to_vertex
